frontend/dockerfile/docs/rules/undefined-var.md

```diff
@@ -13,24 +13,21 @@ Usage of undefined variable '$foo'
 
 ## Description
 
-Before you reference an environment variable or a build argument in a `RUN`
-instruction, you should ensure that the variable is declared in the Dockerfile,
-using the `ARG` or `ENV` instructions.
+This check ensures that environment variables and build arguments are correctly
+declared before being used in the following instructions: `ADD`, `ARG`, `COPY`,
```

Comment:
There are more, like `VOLUME`, `LABEL` for example.

It really is all instructions in https://docs.docker.com/reference/dockerfile/#environment-replacement . Commands like `RUN` that do variable replacement in the specific shell are not analyzed by the check.